File: Ecommerce.Api/Domain/Order.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Ecommerce.Api.Domain;

public enum OrderStatus { Pending, Paid, Failed }

public class Order
{
    [BsonId] public ObjectId Id { get; set; }
    public string OrderNumber { get; set; } = default!;
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "USD";
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/order.ts
export type OrderStatus = 'Pending' | 'Paid' | 'Failed';

export interface Order {
  id: string;
  orderNumber: string;
  amount: number;
  currency: string;
  status: OrderStatus;
  createdAt: string;
}

export interface CreateOrderRequest {
  orderNumber: string;
  amount: number;
  currency?: string;
}
